snapcraft/extensions/_utils.py

```diff
@@ -27,18 +27,14 @@
     from .extension import Extension
 
 
-def apply_extensions(
-    yaml_data: dict[str, Any], *, arch: str, target_arch: str
-) -> dict[str, Any]:
-    """Apply all extensions.
+def apply_extensions(yaml_data: dict[str, Any], *, arch: str, target_arch: str) -> None:
+    """Apply all extensions in place.
```

Comment:
The unit tests are failing for extensions, probably from this change. When we fix those tests, we should double-check how the core22 lifecycle calls this utility.

Reply:
CRAFT-4416